services/motion/builtin/move_request.go

```diff
@@ -0,0 +1,179 @@
+package builtin
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/base/kinematicbase"
+	"go.viam.com/rdk/motionplan"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/motion"
+	"go.viam.com/rdk/spatialmath"
+)
+
+// moveRequest is a structure that contains all the information necessary for to make a move call.
+type moveRequest struct {
+	planRequest        *motionplan.PlanRequest
+	actuator           kinematicbase.KinematicBase
+	execute            func(context.Context, motionplan.Plan) moveResponse
+	position, obstacle *replanner
+}
+
+// plan creates a plan using the currentInputs of the robot and the moveRequest's planRequest.
+func (mr *moveRequest) plan(ctx context.Context) (motionplan.Plan, error) {
+	inputs, err := mr.actuator.CurrentInputs(ctx)
+	if err != nil {
+		return make(motionplan.Plan, 0), err
```

Comment:
should this just return `return nil, err` given we don't expect the caller to use the plan if an error is returned?

Reply:
Yeah idk why I ever did this